src/primitives/types/types_typed.ts

```diff
@@ -121,18 +144,60 @@ export type TypeGenerators = {
   object: any;
   timestamp: any;
 
-  vec2: any;
-  ivec2: any;
-  vec3: any;
-  ivec3: any;
-  vec4: any;
-  ivec4: any;
+  vec2(x?: number, y?: number): Type<Vec2Like, Vector2>;
```

Comment:
do we default values to 0?

Reply:
It depends. There are multiple properties with type `vec2`, and each can have different defaults. Calling `Types.vec2(x, y)` is more like creating a validation instance, with instance-specific defaults.

Using `view3` as an example again (which Caresian and some other nodes use):
```js
// traits.js
view3: {
  position: Types.vec3(),
  quaternion: Types.quat(),
  rotation: Types.vec3(),
  scale: Types.vec3(1, 1, 1),
  eulerOrder: Types.swizzle("xyz"),
},
```

Calling `Types.vec3(a,b,c)` creates a sort of schema that (1) provides a default value of `(a, b, c)` and (2) provides a validator function.
So:
- position's default value is (0, 0, 0) **and** when you set position, any unspecified components default to 0.
- scale's default value is (1, 1, 1) **and** when you set scale, any unspecified components are set to 1.